### ISSUE TITLE
Monthly Guild Stats

### ISSUE THREAD

**Issue by zerinho6** [SwitchbladeBot/switchblade#904]
## Informations that would be displayed:

 -**Member Count**
 -**Role Count**
 -**Channels Count**

 _Additional Ideas..._

 -**Ban Count**
 -**Kick Count**
 -**Active Channels**

### Active Channels

Active Channels would be channels that...

- For text channels:
  If the channel have received 30 or more messages.
- For music channels
  2 or more users(Only 1 bot count) in the channel

Now, the music channel is a bit tricky because trolls could just join the channel and leave and do that for each one, but then..that's not exactly our fault.

## How would it work?

First, the server needs to have atleast x members, I recommend 25 or 50 at max, then someone with MANAGE_GUILD permission would need to enable the module so the bot would put it on a JSON file with other guilds that want the Stats, then, every day the bot would execute a function that would check if the last day checked was today, if not, it would check, for each guild that wants the stats, if it was already updated this month, if not it would write in another file the stats in a format like this:

```JSON
{
    "guildID": {
        "Year": {
            "Month": {
                "memberCount": 0,
                "roleCount": 0,
                "channelCount": 0
            }
        }
    }
}
```

**Comment by fernando457829** [comment]
JSON as database?????? NOOOOOO PLS!!!!!!!!!

**Comment by zerinho6** [comment]
I mean, it's your choice.

**Comment by pedrofracassi** [comment]
I mean, MongoDB serializes stuff in JSON

**Comment by dpaiv0** [comment]
Yeah JSON isn't bad, but with our current configuration, that wouldn't be possible, since dokku just gets the new files from GitHub, deleting anything existant. It's the same procedure that Heroku does.

**Comment by pedrofracassi** [comment]
STORE IT ON THE DATABASE

**Comment by moruzerinho6** [comment]
This shouldn't be done anymore since server insights is now a thing.
